phpunit/src/Glpi/SLMTrait.php

```diff
@@ -0,0 +1,216 @@
+    {
+        $calendar ??= $this->createCalendar();
+
+        return $this->createItem(
+            SLM::class,
+            $data + [
+                'id' => 0,
+                'name' => 'SLM name ' . time(),
+                //                'entities_id' => 0,
+                //                'is_recursive' => 0,
+                'comment' => 'Slm comment text',
+                //                'use_ticket_calendar' => 0,
+                'calendars_id' => $calendar->getID(),
+                //                'date_mod' => null,
+                //                'date_creation' => null,
+            ]
+        );
+    }
+
+    private function createGroup(): \Group
+    {
+        return $this->createItem(\Group::class, [
+            //            'id' => 0,
+            //            'entities_id' => 0,
+            //            'is_recursive' => 0,
+            'name' => 'Group name ' . time(),
+            //            'code' => null,
+            'comment' => 'Group comment text ' . time(),
+            //            'ldap_field' => null,
+            //            'ldap_value' => null,
+            //            'ldap_group_dn' => null,
+            //            'date_mod' => null,
+            //            'groups_id' => 0,
+            //            'completename' => null,
+            //            'level' => 0,
+            //            'ancestors_cache' => null,
+            //            'sons_cache' => null,
+            //            'is_requester' => 1,
+            //            'is_watcher' => 1,
+            //            'is_assign' => 1,
+            //            'is_task' => 1,
+            //            'is_notify' => 1,
+            //            'is_itemgroup' => 1,
+            //            'is_usergroup' => 1,
+            //            'is_manager' => 1,
+            //            'date_creation' => null,
+            //            'recursive_membership' => 0,
+            //            '2fa_enforced' => 0,
+        ]);
+    }
+
+    /**
+     * Calendar with today & tomorrow set a working day (9:00 to 19:00)
+     */
+    private function createCalendar(): \Calendar
+    {
+        $calendar = $this->createItem(\Calendar::class, ['name' => 'Test Calendar ' . time()]);
+        // today
+        $this->createItem(
+            \CalendarSegment::class,
+            [
+                'calendars_id' => $calendar->getID(),
+                'day' => (int) date('w'),
+                'begin' => '09:00:00',
+                'end' => '19:00:00',
+            ]
+        );
+        // tomorrow
+        $this->createItem(
+            \CalendarSegment::class,
+            [
+                'calendars_id' => $calendar->getID(),
+                'day' => (int) date('w') === 6 ? 0 : (int) date('w') + 1, // day of the week number
+                'begin' => '09:00:00',
+                'end' => '19:00:00',
+            ]
+        );
+
+        return $calendar;
+    }
```

Comment:
You may reuse the default calendar (`getItemByTypeName(Calendar::class, 'Default calendar')`).